sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/ProxyUtilTest.java

```diff
@@ -0,0 +1,85 @@
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class ProxyUtilTest {
+    public static final String JAVA_NET_USE_SYSTEM_PROXIES = "java.net.useSystemProxies";
+    public static final String SOME_USERNAME = "some_username";
+    public static final String SOME_PASSWORD = "some_password";
+    public static final String AUTH_TYPE = "BASIC";
+
+    @MethodSource("getProxyConfigurations")
+    @ParameterizedTest
+    public void testProxyOptionsConfiguration(String proxyConfiguration, boolean expectSystemDefault) {
+        Configuration configuration = Configuration.getGlobalConfiguration().clone();
+        configuration = configuration.put(Configuration.PROPERTY_HTTP_PROXY, proxyConfiguration);
+        configuration = configuration.put(ProxyOptions.PROXY_AUTHENTICATION_TYPE, AUTH_TYPE);
+        configuration = configuration.put(ProxyOptions.PROXY_USERNAME, SOME_USERNAME);
+        configuration = configuration.put(ProxyOptions.PROXY_PASSWORD, SOME_PASSWORD);
+
+        ProxyOptions proxyConfig =  ProxyUtil.getDefaultProxyConfiguration(configuration);
+        if (expectSystemDefault) {
+            assertIsSystemDefaultConfig(proxyConfig);
+        } else {
+            Assertions.assertNotNull(proxyConfig);
+            Assertions.assertEquals(ProxyAuthenticationType.BASIC, proxyConfig.getAuthentication());
+            Assertions.assertNotNull(proxyConfig.getProxyAddress());
+            Assertions.assertNotNull(proxyConfig.getCredential());
+            Assertions.assertEquals(SOME_USERNAME, proxyConfig.getCredential().getUserName());
+            Assertions.assertArrayEquals(SOME_PASSWORD.toCharArray(), proxyConfig.getCredential().getPassword());
+        }
+    }
+
+    @Test
+    public void testNullProxyAddress() {
+        Configuration configuration = Configuration.getGlobalConfiguration().clone();
+        configuration = configuration.put(JAVA_NET_USE_SYSTEM_PROXIES, "true");
+        configuration = configuration.put(ProxyOptions.PROXY_AUTHENTICATION_TYPE, AUTH_TYPE);
+        configuration = configuration.put(ProxyOptions.PROXY_USERNAME, SOME_USERNAME);
+        configuration = configuration.put(ProxyOptions.PROXY_PASSWORD, SOME_PASSWORD);
+        ProxyOptions proxyConfig = ProxyUtil.getDefaultProxyConfiguration(configuration);
+        assertIsSystemDefaultConfig(proxyConfig);
+    }
+
+    @Test
+    public void testSystemProxiesConfiguration() {
+        Configuration configuration = Configuration.getGlobalConfiguration().clone();
+        configuration = configuration.put(Configuration.PROPERTY_HTTP_PROXY, "invalid_proxy");
+        configuration = configuration.put(Configuration.PROPERTY_HTTPS_PROXY, "https://localhost:80");
+        configuration = configuration.put(JAVA_NET_USE_SYSTEM_PROXIES, "true");
+
+        ProxyOptions proxyConfig =  ProxyUtil.getDefaultProxyConfiguration(configuration);
+        Assertions.assertNotNull(proxyConfig);
```

Comment:
Should this also assert that it's a system default proxy?

Reply:
Ah, no in this case it would be the core-http-proxy that is configured and returned, but with null creds and non-null address